src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public class zip_LargeFiles : ZipFileTestBase
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+        [OuterLoop("It requires almost 12 GB of free disk space")]
+        public static void UnzipOver4GBZipFile()
+        {
+            byte[] buffer = new byte[1_000_000_000]; // 1 GB
+
+            string zipArchivePath = Path.Combine("ZipTestData", "over4GB.zip");
+            DirectoryInfo tempDir = Directory.CreateDirectory(Path.Combine("ZipTestData", "over4GB"));
+
+            try
+            {
+                for (byte i = 0; i < 6; i++)
+                {
+                    buffer.AsSpan().Fill(i);
```

Comment:
if you're not going to validate the content of the buffers, I guess you can just fill it once and reuse it for each file.